core/iwasm/interpreter/wasm_loader.c

```diff
@@ -8907,12 +8907,24 @@ wasm_loader_emit_br_info(WASMLoaderContext *ctx, BranchBlock *frame_csp,
         /* Part e */
         dynamic_offset =
             frame_csp->dynamic_offset + wasm_get_cell_num(types, arity);
-        ctx->dynamic_offset = dynamic_offset;
         for (i = (int32)arity - 1; i >= 0; i--) {
             cell = (uint8)wasm_value_type_cell_num(types[i]);
             dynamic_offset -= cell;
             emit_operand(ctx, dynamic_offset);
         }
+        /* Update ctx->dynamic_offset and ctx->frame_offset */
```

Comment:
Should we update `ctx->max_dynamic_offset` before this? Otherwise if `frame_csp->dynamic_offset + wasm_get_cell_num(types, arity)` is larger than current `ctx->max_dynamic_offset`, some value may be dropped during the below for loop?